src/olympia/addons/models.py

```diff
@@ -1222,6 +1222,17 @@ def is_featured(self, app=None, lang=None):
         """Is add-on globally featured for this app and language?"""
         return self.id in get_featured_ids(app, lang)
 
+    def get_featured_by_app(self):
+        coll_q = (self.collections.filter(featuredcollection__isnull=False)
+                  .distinct().values_list('featuredcollection__application',
+                                          'featuredcollection__locale'))
+        out = {}
+        for app, locale in coll_q:
+            app_set = out.get(app, set())
+            app_set.add(locale)
+            out[app] = app_set
```

Comment:
that looks like it could or eve should be a `defaultdict`

Reply:
Can you elaborate?  What would the default be, and in what circumstances would having a default be a good thing?

Reply:
`out` could be a defaultdict with a default of `set` so that you end up with something like:

```python
out = defaltdict(set)
out[app].add(locale)
```

Or with a regular dict
 
```python
out = {}
out.setdefault(app, set()).add(locale)
```